js/clinicalTimeline.js

```diff
@@ -156,6 +158,72 @@ window.clinicalTimeline = (function(){
     postTimelineHooks.forEach(function(hook) {
       hook.call();
     });
+
+    var tickTemp = getTickValues(minDays, maxDays, "months");
+    var minDayTick = tickTemp[0];
+    var maxDayTick =  tickTemp[tickTemp.length-1];
+
+    var overview = d3.select(divId).append("svg")
+      .attr("height", 75)
+      .attr("width", width)
+      .attr("class", "overview");
+
+    var xScaleOverview = d3.time.scale()
+      .domain([minDayTick, maxDayTick])
+      .range([0 , width]);
+
+    var xScaleOverviewZoomed = d3.time.scale()
+      .domain([0, -zoomedWidth])
+      .range([0 , width]);
+
+    var xScaleRectangle = d3.time.scale()
+      .domain([0, width])
+      .range([0 , -zoomedWidth]);
+
+    var overviewAxis = d3.svg.axis().scale(xScaleOverview).orient("bottom")
+      .tickFormat(function(d) { 
+        return formatTime(daysToTimeObject(d.valueOf()), "months");
+      })
+      .ticks(tickTemp.length);
+
+    var rectangleOverviewWidth = width/zoomFactor;
+
+    overview.append("g")
+      .attr("class", "x axis")
+      .attr("transform", "translate(0,25)")
+      .call(overviewAxis);
+
+    var dragChart = d3.behavior.drag()
+      .on("drag", function(d,i) {
+        // console.log(d3.transform(d3.select(divId+" svg g").attr("transform")).translate[0], scrolledX, xScaleRectangle(overviewX), overviewX)
+        if(scrolledX){
+          var zoomedBeginTick = xScaleOverviewZoomed(scrolledX); 
+          d3.select(".pos_rectangle").attr("x", zoomedBeginTick);
+          overviewX = zoomedBeginTick;
+        }
+      });
+
+    var dragRectangle = d3.behavior.drag()
+      .on("drag", function(d,i) {
+        var x  = parseInt(d3.select(".pos_rectangle").attr("x"))+d3.event.dx;
+        if(x > 0 && x < width - rectangleOverviewWidth){
+          d3.select(divId+" svg g").attr("transform","translate("+xScaleRectangle(x)+",0)");
+          d3.select(".pos_rectangle").attr("x", x); 
+          overviewX = x;
+          // console.log(d3.transform(d3.select(divId+" svg g").attr("transform")).translate[0])
+        }
+      });
+
+    var rectangle = d3.select(".overview").append("rect")
+      .attr("height", 30)
+      .attr("width", rectangleOverviewWidth)
+      .attr("x", overviewX)
+      .attr("y", 25)
+      .attr("fill", "rgba(25,116,255, 0.4)")
+      .attr("class", "pos_rectangle")
+      .call(dragRectangle);
+    
+    d3.select(divId+" svg").call(dragChart);
```

Comment:
As mentioned the `timeline` function is getting really big, maybe we can move some of this overviewAxis to a separate function/file?